### PR TITLE
Made constraints decrement the system when removed

### PR DESCRIPTION
**Pull request by thegrb93**
Fixes https://github.com/Facepunch/garrysmod-issues/issues/3712

Note: The CallOnRemove must go after the 'SetTable' or it will be overwritten by the new table.

### CODE REVIEW

**Comment by mcNuggets1** [comment]
Didn't even know there was a constraint limit and that it is so incredibly low.

File: garrysmod/lua/includes/modules/constraint.lua [hunk 1]
```diff
@@ -551,6 +522,7 @@ function Rope( Ent1, Ent2, Bone1, Bone2, LPos1, LPos2, length, addlength, forcel
 	if ( IsValid( Constraint ) ) then
```
[the remark this answers is not in the document]
Comment (0fuckthesystem0) [reply garrysmod/lua/includes/modules/constraint.lua]: ok ok, but u need to fix this stuff

[ERROR] lua/includes/modules/constraint.lua:98: attempt to index local 'System' (a nil value)
  1. AddConstraintSystemCount - lua/includes/modules/constraint.lua:98
   2. Rope - lua/includes/modules/constraint.lua:525

[the remark this answers is not in the document]
Comment (thegrb93) [reply garrysmod/lua/includes/modules/constraint.lua]: Oh I see now.

```
-- What the fuck
if ( !Constraint ) then Constraint, rope = rope, nil end
```

That threw me off

[the remark this answers is not in the document]
Comment (thegrb93) [reply garrysmod/lua/includes/modules/constraint.lua]: Should be fixed now. Why was your rope constraint nil anyway?

[the remark this answers is not in the document]
Comment (thegrb93) [reply garrysmod/lua/includes/modules/constraint.lua]: Oh I see. It does that when attaching to the same phys object.

**Comment by thegrb93** [comment]
Added changes to rope code. Will test tonight.

**Comment by robotboy655** [comment]
Decided to fix it a different way:
https://github.com/Facepunch/garrysmod/commit/2f1da1458408965b8c6ec57b1233d3a3cdca88f6

**Pull request closed without merging**